pkg/ottl/ottlfuncs/func_get_xml.go

```diff
@@ -0,0 +1,63 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/antchfx/xmlquery"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type GetXMLArguments[K any] struct {
+	Target ottl.StringGetter[K]
+	XPath  string
+}
+
+func NewGetXMLFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("GetXML", &GetXMLArguments[K]{}, createGetXMLFunction[K])
+}
+
+func createGetXMLFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*GetXMLArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("GetXML args must be of type *GetXMLAguments[K]")
+	}
+
+	if err := validateXPath(args.XPath); err != nil {
+		return nil, err
+	}
+
+	return getXML(args.Target, args.XPath), nil
+}
+
+// getXML returns a XML formatted string that is a result of removing all matching nodes from the target XML.
+// This currently supports removal of elements, attributes, text values, comments, and CharData.
+func getXML[K any](target ottl.StringGetter[K], xPath string) ottl.ExprFunc[K] {
+	return func(ctx context.Context, tCtx K) (any, error) {
+		var doc *xmlquery.Node
+		if targetVal, err := target.Get(ctx, tCtx); err != nil {
+			return nil, err
+		} else if doc, err = parseNodesXML(targetVal); err != nil {
+			return nil, err
+		}
+
+		nodes, err := xmlquery.QueryAll(doc, xPath)
+		if err != nil {
+			return nil, err
+		}
+
+		result := &xmlquery.Node{Type: xmlquery.DocumentNode}
+		for _, n := range nodes {
+			if n.Type != xmlquery.ElementNode {
+				continue
+			}
+			xmlquery.AddChild(result, n)
+		}
+		return result.OutputXML(false), nil
```

Comment:
parsing from a string and returning a string gives use the most flexibility in how the converters are used, but I am a little worried about performance: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35281#issuecomment-2379644212